src/HttpHeader.cc

```diff
@@ -522,18 +522,37 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
                Raw("header", header_start, hdrLen));
     }
 
+    String rawTe;
     if (clen.prohibitedAndIgnored()) {
         // RFC 7230 section 3.3.2: A server MUST NOT send a Content-Length
         // header field in any response with a status code of 1xx (Informational)
         // or 204 (No Content). And RFC 7230 3.3.3#1 tells recipients to ignore
         // such Content-Lengths.
         if (delById(Http::HdrType::CONTENT_LENGTH))
             debugs(55, 3, "Content-Length is " << clen.prohibitedAndIgnored());
-    } else if (chunked()) {
+
+        // RFC 7230 section 3.3.1 has the same criteria forbid Transfer-Encoding
```

Comment:
The missing GET/HEAD criteria cases are already XXX marked in the ContentLengthInterpreter class.

The addition of trailer as a place where these header are forbidden is I believe an oversight in the RFC texts. This handling seems correct to me for that case. Will add a note comment about that to the caller.

I am not seeing any other ways that method can return true/false result different to the RFC criteria.

Reply:
> I am not seeing any other ways that method can return true/false result different to the RFC criteria.

It sounds like you are focusing on the details of the method implementation but forgetting that the method itself is a part of the class that is completely _unused_ for Transfer-Encoding header fields: ContentLengthInterpreter is only used for Content-Length fields. Please reread my original change [request](https://github.com/squid-cache/squid/pull/702/files#r463819674) with that fact in mind.

> Added a note comment about that in b1f897d .

FYI: The new note/comment is misplaced: prohibitedAndIgnored() _caller_ should not care why or how that "prohibit and ignore" decision was made. The same can be said about the old/existing `RFC 7230 section 3.3.2: A server MUST NOT` comment. All this information should most likely go into ContentLengthInterpreter instead of its users. Some of it may be there already. This caller should just apply the already made decision.

Reply:
Yes I am focusing on the implementation for this PR. I looked into the interpreter usage and saw that the request-method / reply-status criteria used by prohibitedAndIgnored() are set by callers of the parser. So this particular check can be used regardless of whether the Content-Length header exists, it is the other checks that depend on C-L header existence.

If the caller lacks the method/status info for this check it cannot pass them in for T-E filtering anyway - so I think this is sufficient (for now) to prune away T-E headers whenever this check returns true.